pkg/mimirpb/compat_test.go

```diff
@@ -564,14 +628,23 @@ func TestPrometheusSampleHistogramInSyncWithMimirPbSampleHistogram(t *testing.T)
 	protoType := reflect.TypeOf(SampleHistogram{})
 	prometheusType := reflect.TypeOf(model.SampleHistogram{})
 
-	test.RequireSameShape(t, prometheusType, protoType)
+	test.RequireSameShape(t, prometheusType, protoType, false)
 }
 
-// Check that Promtheus Label and MimirPb LabelAdapter types converted
+// Check that Prometheus Label and MimirPb LabelAdapter types converted
 // into each other with unsafe.Pointer are compatible
 func TestPrometheusLabelsInSyncWithMimirPbLabelAdapter(t *testing.T) {
 	protoType := reflect.TypeOf(LabelAdapter{})
 	prometheusType := reflect.TypeOf(labels.Label{})
 
-	test.RequireSameShape(t, prometheusType, protoType)
+	test.RequireSameShape(t, prometheusType, protoType, false)
```

Comment:
This entire test can be checked in compile time:
```
	_ = labels.Label(LabelAdapter{})
```

Reply:
I'd guess so, but that's more strict than we need. I'm assuming you're taking advantage of https://go.dev/ref/spec#Conversions, https://go.dev/ref/spec#Assignability

Reply:
@charleskorn What do you think?

Reply:
Happy either way, if we want to use Oleg's trick we should add a comment explaining what we're trying to achieve and that it's more strict than we really need.